.github/workflows/os-llvm-sycl-build.yml

```diff
@@ -0,0 +1,84 @@
+name: Build with Open Source LLVM-Sycl compiler
+on:
+  pull_request:
+  push:
+    branches: [master]
+
+jobs:
+  install-compiler:
+    name: Install Nightly build of DPC++ toolchain
+    runs-on: ubuntu-20.04
+
+    env:
+      OCLCPUEXP_FN: oclcpuexp-2021.12.6.0.19_rel.tar.gz
+      FPGAEMU_FN: fpgaemu-2021.12.6.0.19_rel.tar.gz
+      TBB_FN: oneapi-tbb-2021.4.0-lin.tgz
+
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Download nightly and components
+        shell: bash -l {0}
+        run: |
+          cd /home/runner/work
+          mkdir -p sycl_bundle
+          cd sycl_bundle
+          git clone http://github.com/intel/llvm.git llvm_src
+          cd llvm_src
+          export NIGHTLY_TAG=$(python3 -c "import sys, urllib.parse as ul; print (ul.quote_plus(sys.argv[1]))" $(git tag | sort -V | tail -1))
+          cd ..
+          export DOWNLOAD_URL_PREFIX=https://github.com/intel/llvm/releases/download
+          wget ${DOWNLOAD_URL_PREFIX}/${NIGHTLY_TAG}/dpcpp-compiler.tar.gz
```

Comment:
We can cache these downloaded copies to speed up the workflow.

Reply:
This would be great. I just need to know where to look for an example. GTest installation was being cached, so I will look there.

Reply:
You do not need a conf file. As we are storing the cloned repo we might as well run `git tag | sort -V | tail -1` and compare the SHA with what the command I pasted returns.